### PR TITLE
WSL: More consistently log when child processes fail.

### LINKED ISSUE #658: Always wrap calls to wsl in a try/catch block to show errors
Often, an error message resulting from a call to `wsl` doesn't give enough information, as in `The system could not find the specified file`. 

If we at least emit the command with arguments when this happens, that will help troubleshoot problems, for both developers and support people.

### PR DESCRIPTION
**Pull request by mook-as**
Fixes #658 — though I may have additional PRs for #538 and related.

### CODE REVIEW

**Comment by mattfarina** [comment]
@mook-as can you please rebase this

**Review by mattfarina — COMMENTED** [review]
@mook-as There appears to be a problem writing logs. For example, with this PR I get nothing in my wsl.log file until I exit the application. Then the full contents of the log are:

```
K3s exited gracefully.
agent exited with status 1 signal null
```

All of the details about copying certs, etc are missing but appear on main. This happens for other log files as well.

Note, this is on Windows.

**Comment by mook-as** [comment]
> @mook-as There appears to be a problem writing logs. For example, with this PR I get nothing in my wsl.log file until I exit the application. 

Hmm, I could not reproduce this (even in a packaged build).  However, I _did_ fix the PR up, to ensure that the logging directory is created before attempting to add logs to them (otherwise we get a string of dialog boxes about not being able to write logs).

**Review by mattfarina — COMMENTED** [review]
@mook-as it's working now and looks good. Except for the lint issue

**Comment by ericpromislow** [comment]
As long as we're revisiting Windows logs, I have to make sure I'm not tailing the logs in the Ubuntu WSL subsystem when I start RD up. Otherwise RD complains that it can't open a file for writing. I can't repro that by running `tail` in WSL and then in a node session in a powershell terminal running `x = fs.createStreamLog('wsl.log', { mode: 0o600 }); x.write('some stuff\n'); x.close()` so there's more going on.

**Review by ericpromislow — DISMISSED** [review]
Just a typo or two, and a request for a comment on how the logging proxy works.

File: src/k8s-engine/wsl.ts [hunk 1]
```diff
@@ -317,7 +325,23 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     const mountRoot = '/mnt/wsl/rancher-desktop/run/data';
 
     await this.execCommand('mkdir', '-p', mountRoot);
-    await this.execWSL('--distribution', DATA_INSTANCE_NAME, 'mount', '--bind', '/', mountRoot);
+    // Only bind mount the root if it doesn't exist; because this is in the
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/k8s-engine/wsl.ts:328]: It was a change that was blocked on the change to `execWSL`.

**Pull request merged**